server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -91,6 +93,7 @@ public class ContextIndexSearcher extends IndexSearcher implements Releasable {
     // don't create slices with less than this number of docs
     private final int minimumDocsPerSlice;
 
+    private final Map<Thread, Boolean> timeoutOverwrites = new ConcurrentHashMap<>();
```

Comment:
Can this be a set ? Don't we always set the boolean to true?

Reply:
yes